applications/registration-api/src/main/java/no/dcat/controller/CatalogController.java

```diff
@@ -74,24 +77,19 @@ public CatalogController(CatalogRepository catalogRepository, SpringSecurityCont
     public PagedResources<Resource<Catalog>> listCatalogs(Pageable pageable, PagedResourcesAssembler<Catalog> assembler) {
 
         Authentication auth = springSecurityContextService.getAuthentication();
+        ResourceRoles resourceRoles = ResourceRoles.fromAuthorities(auth.getAuthorities());
 
-        Set<String> validCatalogs = new HashSet<>();
 
-        for (GrantedAuthority authority : auth.getAuthorities()) {
-            String authorityString = authority.getAuthority();
-            String publisher = StringUtils.substringBetween(authorityString, "publisher:", ":");
-            if (isNotEmpty(publisher)) {
-                validCatalogs.add(publisher);
-            }
-        }
+        Set<String> adminableOrgNrs = resourceRoles.getPublishersForAdminPermission();
```

Comment:
Method name: it's not really getting a permission, as much as a list of organisations that can be administrated

Reply:
jeg vil jo si "List of Organisations that i have admin permission to".  Til å konvertere dt til verb, ligger en til abstraksjonslag til.
I frontend, Øyvind sitt kode har modellert "ability". Vi kan modellere samme måte, ability er som verb. Vet ikke :)

Reply:
vi er på en måte knuttet abstraksjoner Spring Security setter oss. så vi har konsepter som ressurs, og permission

Reply:
liker du getPublishersICanAdmin bedre?

Reply:
vi kommet enige til løsning:` getPublishersForPermission(PublisherPermission publisherPermission)`